Add formatting check

julianneswinoga opened the pull request:
Style was loosely based off of the Microsoft style (started with `clang-format --style=Microsoft --dump-config > .clang-format` and made changes)
Look over your favorite files and let me know if you'd like to propose any style changes @OpenAstroTech/developer 's :smile: 

See #111 for work-in-progress PR, as well as what @openastrotech-bot would comment if the check fails

julianneswinoga committed:
Add .clang-format file, add GitHub actions code style check

- Change workflows so that they run on all PRs, not just targeted at develop
- Update README.md for code style

julianneswinoga commented:
The reason that the check doesn't run on this PR is because of how `pull_request_target` works (only runs if that workflow file has not been modified by the PR), and we need that if we want the workflow to have read/write permissions, which we need in order to tell the bot to add a comment if the check fails :sweat_smile:

ClutchplateDude commented on Configuration_sample_local.hpp:
```diff
@@ -27,33 +27,36 @@
-#define DEC_STEPPER_ACCELERATION   600   // Defaults: 600 for 28BYJ-48, 6000 for NEMA17
+#define RA_DRIVER_TYPE DRIVER_TYPE_ULN2003
+#define DEC_DRIVER_TYPE DRIVER_TYPE_ULN2003
+#define RA_STEPPER_SPEED                                                                                               \
```
What is this mess it's doing? Splitting it over two lines? No more lining up values? Booo, this needs to change

julianneswinoga replied:
Hmm, yeah this seems to be because of the inline comment

ClutchplateDude commented on src/DayTime.cpp:
```diff
@@ -14,337 +14,306 @@
+    long sgn = 1;
+    LOGV2(DEBUG_MEADE, F("DayTime: Parse Coord from [%s]"), s.c_str());
+    // Check whether we have a sign. This should be able to parse RA and DEC strings (RA never has a sign, and DEC should always have one).
+    if ((s[i] == '-') || (s[i] == '+')) {
```
Opeing brace on seperate new line, please.

julianneswinoga replied:
I vastly prefer the opening brace for control flow to be on the same line, but I think we should have a team vote for this point :grin:

ClutchplateDude commented on src/EPROMStore.cpp:
```diff
@@ -156,161 +158,168 @@ void EEPROMStore::displayContents()
+    unsigned result = (MAGIC_MARKER_VALUE | item);
+    bool res        = ((marker & check) == result);
+    LOGV6(DEBUG_EEPROM,
+          F("EEPROM: IsDataPresent (%x). Checking (%x & %x) == %x => %d"),
```
Urgh. What makes this go over multiple lines? Can we make it stay on one line?

julianneswinoga replied:
It splits the lines up because of the character limit (120), lets check back on this when we bump that up

ClutchplateDude commented on src/EPROMStore.cpp:
```diff
@@ -319,450 +328,424 @@ void EEPROMStore::updateFlagsExtended(ExtendedItemFlag item)
-  return brightness; // dimensionless scalar
+    byte brightness = readUint8(LCD_BRIGHTNESS_ADDR);
+    if (brightness == 0)
+        brightness = 10;  // Have a reasonable minimum in case nothing is stored
```
Is there a rule available that enforces braces on all if()s?

julianneswinoga replied:
No, `clang-format` won't alter the code like this but `clang-tidy` has that option. We could look at some `clang-tidy` integration in the future!

ClutchplateDude replied:
I don't mean for it to change it, just to flag it.

ClutchplateDude commented on src/EPROMStore.hpp:
```diff
@@ -8,164 +8,188 @@
+        EXTENDED_FLAGS_ADDR         = 21,
+        _EXTENDED_FLAGS_ADDR_1      = 22,  // Uint16
+        RA_PARKING_POS_ADDR         = 23,
+        _RA_PARKING_POS_ADDR_1,
```
Probably should add the explicit numbers here, so they format better.

julianneswinoga replied:
Yeah, or rearrange the enum such that they're in the correct order. We could also add some `static_assert()`s to double check that no-one breaks the order unintentionally

ClutchplateDude commented on .clang-format:
```diff
@@ -0,0 +1,148 @@
+Language:        Cpp
+AccessModifierOffset: -2
+AlignAfterOpenBracket: Align
+AlignConsecutiveMacros: false
```
I think this is the one that should be true.

ClutchplateDude commented on .clang-format:
```diff
@@ -0,0 +1,148 @@
+BreakConstructorInitializers: BeforeColon
+BreakAfterJavaFieldAnnotations: false
+BreakStringLiterals: true
+ColumnLimit:     120
```
Let's up this to 150

julianneswinoga replied:
I think 150 is a bit much, but yeah the 'standard' 120 is a bit too restrictive I think. 140?

ClutchplateDude replied:
Sure

ClutchplateDude commented on src/c65_startup.hpp:
```diff
@@ -28,178 +29,171 @@ enum startupState_t {
-    case StartupSetRoll : {
-      inStartup = false;
-      LOGV1(DEBUG_INFO, F("STARTUP: Switching to CAL menu!"));
+        #if USE_GYRO_LEVEL == 1
```
This is a weird indentation. And in some files #if and #ifdefs don't indent at all?

julianneswinoga replied:
What's the weird indentation? The preprocessor instructions are indented separately than the actual code

ClutchplateDude replied:
Oh. Interesting.

ClutchplateDude commented on .clang-format:
```diff
@@ -0,0 +1,148 @@
+    SortPriority:    0
+IncludeIsMainRegex: '(Test)?$'
+IncludeIsMainSourceRegex: ''
+IndentCaseLabels: false
```
This should be true

julianneswinoga replied:
I don't agree, in my mind that leads to way too much indentation cruft since the case body is always indented 1 more than the case label.

ClutchplateDude reviewed (commented):
Looked over some of the files to see what it's doing.... needs some tweaks, but I like having it be consistent over the code base! Thanks.

ClutchplateDude commented on src/DayTime.cpp:
```diff
@@ -14,337 +14,331 @@
         {
-          strcpy(p, achDegs);
-          p += strlen(achDegs);
+        case '{': {
```
This opening curly needs to be on a new line.

openastrotech-bot committed:
Reformat codebase according to .clang-format

Using script:
bash -c 'shopt -s nullglob; for i in ./{.,src,src/inc,boards/*}/*.{c,cpp,h,hpp}; do clang-format -i $i; done'
